source/vir/src/recursion.rs

```diff
@@ -459,9 +459,11 @@ pub(crate) fn expand_call_graph(
     // Add D: T --> f and D: T --> T where f is one of D's methods that implements T
     if let FunctionKind::TraitMethodImpl { trait_path, datatype, .. } = function.x.kind.clone() {
         let t_node = Node::Trait(trait_path.clone());
-        let dt_node = Node::DatatypeTraitBound { datatype, trait_path };
-        call_graph.add_edge(t_node, dt_node.clone());
-        call_graph.add_edge(dt_node, f_node.clone());
```

Comment:
I think `DatatypeTraitBound` should be generalized to allow non-`Datatype` types.  So `TypX::Datatype` would not be a special case here.